recipes/nettle/all/conanfile.py

```diff
@@ -0,0 +1,87 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class NettleTLS(ConanFile):
+    name = "nettle"
+    description = "The Nettle and Hogweed low-level cryptographic libraries"
+    homepage = "https://www.lysator.liu.se/~nisse/nettle"
+    topics = ("conan", "nettle", "crypto", "low-level-cryptographic", "cryptographic")
+    license = ("GPL-2.0-or-later", "GPL-3.0-or-later")
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "public_key": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "public_key": True,
+    }
+
+    _source_subfolder = "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def requirements(self):
+        if self.options.public_key:
+            self.requires("gmp/6.1.2")
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        if self.settings.compiler == "Visual Studio":
+            raise ConanInvalidConfiguration("Nettle cannot be built using Visual Studio")
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("nettle-{}".format(self.version), self._source_subfolder)
+
+    def build_requirements(self):
+        if tools.os_info.is_windows and not "CONAN_BASH_PATH" in os.environ:
+            self.build_requires("msys2/20190524")
+
+    def _configure_autotools(self):
+        autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
```

Comment:
I suggest using a cache for autotools object, otherwise, this recipe will run configure step twice, one when building and another for installing. For instance https://github.com/conan-io/conan-center-index/blob/ded36f1ee8bc22360872e43853f3e9cb586911d5/recipes/nasm/ALL/conanfile.py#L43